src/framework-configuration.js

```diff
@@ -256,17 +303,37 @@ export class FrameworkConfiguration {
   /**
    * Configures an external, 3rd party plugin before Aurelia starts.
    * @param plugin The ID of the 3rd party plugin to configure.
-   * @param config The configuration for the specified plugin.
+   * @param pluginConfig The configuration for the specified plugin.
    * @return Returns the current FrameworkConfiguration instance.
  */
-  plugin(plugin: string, config?: any): FrameworkConfiguration {
+  plugin(
+    plugin: string | ((frameworkConfig: FrameworkConfiguration) => any) | FrameworkPluginInfo,
+    pluginConfig?: any
+  ): FrameworkConfiguration {
     assertProcessed(this);
 
-    if (typeof (plugin) === 'string') {
-      return this.plugin({ moduleId: plugin, resourcesRelativeTo: [plugin, ''], config: config || {} });
+    let info: FrameworkPluginInfo;
+    switch (typeof plugin) {
+    case 'string':
+      info = { moduleId: plugin, resourcesRelativeTo: [plugin, ''], config: pluginConfig || {} };
+      break;
+    case 'function':
+      if (this.queuedPlugins.indexOf(plugin) !== -1) {
```

Comment:
Basically, you want `queuedPlugins` to be a `Set`, why not use one?
Note: `Set.prototype.forEach` enumerates values in insertion order.

Reply:
Here is the small discussion that got it moved from `WeakSet` to `Set` https://github.com/aurelia/framework/pull/885#discussion_r187905475 . I was mainly doing this for IE11 users, who want to deliver Aurelia app without having to use collection polyfills, as after this feature set, it will be easy to bootstrap / create app without importing polyfills / configure it in webpack plugin so someone may actually wants to do so.